SteamKit2/SteamKit2/Util/Utils.cs

```diff
@@ -363,42 +365,95 @@ public static uint GetIPAddress( IPAddress ipAddr )
             return BitConverter.ToUInt32( addrBytes, 0 );
         }
 
-
-        public static uint EndianSwap( uint input )
+        public static IPAddress GetIPAddress( CMsgIPAddress ipAddr )
         {
-            return ( uint )IPAddress.NetworkToHostOrder( ( int )input );
+            if ( ipAddr.ShouldSerializev6() )
+            {
+                return new IPAddress( ipAddr.v6 );
+            }
+            else
+            {
+                return GetIPAddress( ipAddr.v4 );
+            }
         }
-        public static ulong EndianSwap( ulong input )
+
+        public static CMsgIPAddress GetMsgIPAddress( IPAddress ipAddr )
         {
-            return ( ulong )IPAddress.NetworkToHostOrder( ( long )input );
+            var msgIpAddress = new CMsgIPAddress();
+            byte[] addrBytes = ipAddr.GetAddressBytes();
+
+            if ( ipAddr.AddressFamily == AddressFamily.InterNetworkV6 )
+            {
+                msgIpAddress.v6 = addrBytes;
+            }
+            else
+            {
+                Array.Reverse( addrBytes );
+
+                msgIpAddress.v4 = BitConverter.ToUInt32( addrBytes, 0 );
+            }
+
+            return msgIpAddress;
         }
-        public static ushort EndianSwap( ushort input )
+
+        public static CMsgIPAddress ObfuscatePrivateIP( IPAddress ipAddr )
         {

```

Comment:
I'd rather have two methods - one to convert `IPAddress` into `CMsgIPAddress`, and one to apply obfuscation to `CMsgIPAddress`